cmd/topazd/topaz_run.go

```diff
@@ -82,29 +84,31 @@ var cmdRun = &cobra.Command{
 				topazApp.Configuration.ControllerConfig,
 				app.KeepAliveDialOptionsProvider(),
 			)
-
 			runtime, runtimeCleanup, err := topaz.NewRuntimeResolver(
 				topazApp.Context,
 				topazApp.Logger,
 				topazApp.Configuration,
 				controllerFactory,
 				decisionlog,
 				directory,
+				topazApp,
```

Comment:
If we change SetServiceStatus to be a func instead of a method, we do not need any changes to the main to pass the app pointer around.